### PR TITLE
Storage -> Memory for "Out of Memory" error message

### PR DESCRIPTION
**Pull request by benaadams**
Seen in https://github.com/dotnet/core/issues/1192

Also in https://github.com/dotnet/corefx/issues/24139

Both `System.Net.Http.WinHttpException`s

### CODE REVIEW

**Comment by benaadams** [comment]
Also in https://github.com/dotnet/corefx/issues/24139

File: src/pal/src/misc/errorstrings.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ ErrorString palErrorStrings[] =
     { ERROR_TOO_MANY_OPEN_FILES, W("The system cannot open the file.\n") },
     { ERROR_ACCESS_DENIED, W("Access is denied.\n") },
     { ERROR_INVALID_HANDLE, W("The handle is invalid.\n") },
-    { ERROR_NOT_ENOUGH_MEMORY, W("Not enough storage is available to process this command.\n") },
+    { ERROR_NOT_ENOUGH_MEMORY, W("Not enough memory is available to process this command.\n") },
```
Comment (jkotas) [inline src/pal/src/misc/errorstrings.cpp]: These strings are used on Unix only. The messages are provided by the `FormatMessage` API on Windows. I do not think that this change will do anything to the message in the linked issues. Both of them were on Windows.

**Comment by danmosemsft** [comment]
Some of these BTW the PAL is never going to need eg ERROR_EVENTLOG_FILE_CORRUPT

File: src/pal/src/misc/errorstrings.cpp (hunk 1 above)
Comment (benaadams) [reply src/pal/src/misc/errorstrings.cpp]: Suppose its technically correct then :-/
```
ERROR_NOT_ENOUGH_MEMORY
8 (0x8)
Not enough storage is available to process this command.
```

**Pull request closed without merging**